kb_sdk_plus.md

```diff
@@ -1,10 +1,98 @@
 # KB-SDK plus
 
+## Why?
+
+The KBase SDK is a core piece of infrastructure, but it has become exceedingly difficult to
+maintain and expand as the test suite is exceptionally complex and currently does not pass.
+As such, large changes to the codebase are risky and no updates have been released for 5+ years.
+Furthermore, due to outdated dependencies, the SDK will not run on modern machines.
+
+To rectify this while still supporting older apps, we have elected to fork the
+https://github.com/kbase/kb_sdk repo to `kb_sdk_plus`. Version 1.2.1 is the last `kb_sdk`
+release and will be kept as is. Future SDK development will occur in `kb_sdk_plus`.
+
+## Why not kb_sdk2?
+
+`<repo>2` generally signifies a greenfield rewrite of a codebase in KBase,
+which `kb_sdk_plus` is not. It's forking a codebase to leave the original codebase untouched
+and runnable while we make changes to the forked codebase.
+
+## Guidelines
+
+* Backwards incompatible changes are expected in some respects:
+  * R code generation will be removed
+    * Unused in KBase or anywhere else as far as we know other than one test app
+  * Perl code generation will almost certainly be removed
+    * Perl apps can continue to be supported with the original SDK, or wrapped with
+      Python / Java
+      * https://github.com/boriel/perlfunc for instance
```

Comment:
Any examples of this in KBase? I don't think there are, just checking if anyone's tried that.

Reply:
I'm unaware of any